runtime/connectors/s3/s3.go

```diff
@@ -122,8 +130,9 @@ func (c connector) ConsumeAsIterator(ctx context.Context, env *connectors.Env, s
 	}
 
 	it, err := rillblob.NewIterator(ctx, bucketObj, opts)
-	if gcerrors.Code(err) == gcerrors.PermissionDenied && creds != credentials.AnonymousCredentials {
-		// s3 throws permission denied error in case we are trying to access public buckets and passing some credentials
+	errCode := gcerrors.Code(err)
+	if (errCode == gcerrors.PermissionDenied || errCode == gcerrors.Unknown) && creds != credentials.AnonymousCredentials {
+		// s3 throws error in case we are trying to access public buckets and passing some credentials
```

Comment:
Responding to your other comment about error codes. I don't think we should use `IsAccessible` since it's an extra RPC. But this check looks a little arbitrary, please document it (and check that it covers the relevant cases) (and also, let's add that `err != nil` check now, even if it might be redundant – I find myself wondering if `gcerrors.Code` might return `Unknown` for nil errors).